src/internal/memory/memory_block.hpp

```diff
@@ -105,25 +105,25 @@ struct memory_block_compare_size
     }
 };
 
-template <typename Compare = std::less<>>
-struct memory_block_compare_addr
+template <typename CompareT = std::less<>>
+struct MemoryBlockCompareAddr
```

Comment:
- Not covered by any unit tests
- Doesn't appear to be used in the code base.

Reply:
do tests need direct usage and expects to get marked as covered.

```
    auto ucx_block = ucx->registration_cache().lookup(md.data());

    CHECK(ucx_block.local_handle());
    CHECK(ucx_block.remote_handle());
    CHECK(ucx_block.remote_handle_size());
```

the above lines perform the address compare as part of the BlockManager lookup which is part of the RegistrationCache resulting in a valid block returned.

Reply:
this was used for a specific type of memory resource that has not been ported back into srf.  i'll note it and we can remove it if we don't need to re-enabled that resource type.  i think rmm's arena resource will replace the old best_fit (bfit_allocator) from the old code base.

Reply:
Will resolve with #109